source/acquaman/AMScanController.h

```diff
@@ -65,147 +84,403 @@ class AMScanController : public QObject
+	  */
 	virtual bool initializeImplementation() = 0;
-	/// Implement to start a scan (ie: transition from Initialized to Running).  Returns whether or not the scan can be started, not that the scan has started. After the scan is running, call setStarted().
+
+	/*!
+	  * Performs the actions required to start a scan running.
+	  *
+	  * In the base class this is a pure virtual function.
+	  * \returns True if the transition to running can occur, false otherwise.
+	  * NOTE: Returns whether or not the scan can be started, not that the scan
+	  * has started. Pay attention to the started() signal to determine when a scan
+	  * starts.
+	  */
 	virtual bool startImplementation() = 0;
-	/// Implement to define whether or not your scan controller is capable of pausing (defaults to not able to pause). If you canPause (return true), then you need to implement a pauseImplementation and a resumeImplementation.
+
+	/*!
+	  * Whether or not the scan controller is capable of pausing. Implementing
+	  * classes should alter this from the default value of false if scan controller
+	  * is capable of pausing.
+	  */
 	virtual bool canPause() const { return false; }
-	/// Implement to pause a running scan (ie: transition from Running to Paused).  After the scan is paused, call setPaused(). (Note: If it's not possible to pause this type of scan (ie: canPause() is false), this will never be called.)
+
+	/*!
+	  * Performs the actions required to pause a scan. Implementing classes which
+	  * override the canPause() function to indicate that their scan is capable
+	  * of pausing must re-implement this function.
+	  */
 	virtual void pauseImplementation() {}
-	/// Implement to resume a paused scan (ie: transition from Paused to Running).  If pauseImplementation() returns false, you don't need to do anything here. After the scan is resumed, call setResumed().
+
+	/*!
+	  * Performs the actions required to resume a scan from the paused state. Implementing
+	  * classes which override the canPause() function to indicate that their scan
+	  * is capable of pausing must re-implement this function.
+	  */
 	virtual void resumeImplementation() {}
-	/// Implement to cancel a scan. This could be called from any state.  After the scan is cancelled, call setCancelled().
+
+	/*!
+	  * Performs the actions required to cancel a scan.
+	  * NOTE: This could be called from any state.
+	  *
+	  * In the case class this is  pure virtual function.
+	  */
 	virtual void cancelImplementation() = 0;
-	/// Implement to stop a scan.  This can only be called from the running or paused state.  This will be responsible for calling setFinished() or ensuring that it is called.  Pure-virtual because if you can cancel, then you can also stop.  This is defaulted to an empty string because you may not need to distinguish between reasons.
+
+	/*!
+	  * Performs the actions required to stop a scan.
+	  * \param command ( = "" ) ~ An optional reason for stopping the scan.
```

Comment:
The command string is to allow a general stopping mechanism.  There could be a variety of reasons why/how you want to stop and having a string encapsulate that is the most general way.